In [1]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/telco_customer_churn.csv')

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7043 rows x 21 columns]

### Welcome to Data Manipulation Homework. Please answer the questions below!

In [4]:
syarat = df['tenure'] != 0
df_baru = df[syarat]
df_baru.shape

(7032, 21)

1. Buanglah baris dengan `tenure` samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan `tenure` TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

2. Bagi customer yang ***TIDAK MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa ***rata-rata*** dari `tenure` mereka?

3. Bagi customer yang ***MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa maksimal `MonthlyCharges` yang tercata?

4. Berapa banyak customer yang ***TIDAK MEMILIKI PARTNER*** namun ***MEMILIKI DEPENDENTS***?

5. Berapa rata-rata ***TOTAL CHARGES*** dari customer ***NON-SENIOR CITIZEN*** (Senior Citizen = 0) yang memiliki ***PHONE SERVICE*** dan memiliki ***INTERNET SERVICE FIBER OPTIC***?

#### Silakan menuliskan kode untuk menjawab 5 pertanyaan di atas...

In [ ]:
#soal 2
df_2=df_baru.copy()
syarat_1=df_2['Partner'] == "No"
syarat_2=df_2['Dependents'] == "No"

df_2h=df_2[syarat_1 & syarat_2]


In [ ]:

df_2h.mean()

C:\Users\MYWINDOWS\AppData\Local\Temp\ipykernel_17648\1544642153.py:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df_2h.mean()


SeniorCitizen      0.171037
tenure            23.303354
MonthlyCharges    62.983735
dtype: float64

In [ ]:
#soal 3
df_3=df_baru.copy()
syarat_3=df_3['Partner'] == "Yes"
syarat_2=df_3['Dependents'] == "No"

df_3h=df_3[syarat_3 & syarat_2]

In [ ]:
df_3h['MonthlyCharges'].max()

118.6

In [24]:
#soal 4
df_4=df_baru.copy()
syarat_a=df_4['Partner'] == "No"
syarat_b=df_4['Dependents'] == "Yes"

df_4h=df_4[syarat_a & syarat_b]

In [23]:
df_4h['customerID'].count()

361

In [21]:
len(df_4h['customerID'])

0

In [ ]:
#soal 5
df_5=df_baru.copy()
syarat_5=df_5['SeniorCitizen'] == 0
syarat_6=df_5['PhoneService'] == "Yes"
syarat_7=df_5['InternetService'] == "Fiber optic"

df_5h=df_5[syarat_5 & syarat_6 & syarat_7]

In [ ]:
df_5h.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2265 entries, 4 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        2265 non-null   object 
 1   gender            2265 non-null   object 
 2   SeniorCitizen     2265 non-null   int64  
 3   Partner           2265 non-null   object 
 4   Dependents        2265 non-null   object 
 5   tenure            2265 non-null   int64  
 6   PhoneService      2265 non-null   object 
 7   MultipleLines     2265 non-null   object 
 8   InternetService   2265 non-null   object 
 9   OnlineSecurity    2265 non-null   object 
 10  OnlineBackup      2265 non-null   object 
 11  DeviceProtection  2265 non-null   object 
 12  TechSupport       2265 non-null   object 
 13  StreamingTV       2265 non-null   object 
 14  StreamingMovies   2265 non-null   object 
 15  Contract          2265 non-null   object 
 16  PaperlessBilling  2265 non-null   object 


In [ ]:
df_5h['TotalCharges'] = df_5h['TotalCharges'].astype('float')

C:\Users\MYWINDOWS\AppData\Local\Temp\ipykernel_17648\214004989.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_5h['TotalCharges'] = df_5h['TotalCharges'].astype('float')


In [ ]:
round(df_5h['TotalCharges'].mean(),2)

3226.49

### Silakan menjawab di bawah ini:

#### Soal 1:
Buanglah baris dengan tenure samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan tenure TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

#### Jawaban: 7032 (langsung edit di sini aja)

#### Soal 2:
Bagi customer yang TIDAK MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa rata-rata dari tenure mereka?

#### Jawaban: 23.303354

#### Soal 3:
Bagi customer yang MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa maksimal MonthlyCharges yang tercata?

#### Jawaban: 118.6

#### Soal 4:
Berapa banyak customer yang TIDAK MEMILIKI PARTNER namun MEMILIKI DEPENDENTS?

#### Jawaban: 361

#### Soal 5:
Berapa rata-rata TOTAL CHARGES dari customer NON-SENIOR CITIZEN (Senior Citizen = 0) yang memiliki PHONE SERVICE dan memiliki INTERNET SERVICE FIBER OPTIC?

#### Jawaban: 3226.49